app/commands/cart_services/create_reservation_from_cart_item.rb

```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+#
+# Copyright 2021 Victoria Garcia
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+class CartServices::CreateReservationFromCartItem
+  attr_reader :cart_item, :customer
+
+  def initialize(cart_item, customer)
+    @cart_item = cart_item
+    @customer = customer
+  end
+
+  def call
+    return nil if !@cart_item.acquirable.kind_of?(Membership)
```

Comment:
Why can't we create a cart item from anything other than a `Membership`?

Reply:
Because we don't have anything right now to create one from? I've tried to hold space for an 'unknown-' type CartItem, but I didn't want to hold too much space, because I didn't want unknown-type CartItems to make it out into production.

Reply:
In case it wasn't clear-- The ability to create diffferent kinds of CartItems is the whole reason behind having Benefitable, Acquirable, and Holdable:  Holdable allows for the creation of different kinds of electronic (site selection tokens, reservations, etc) through interaction with the Cart.  Acquirable allows for different kinds of things to be bought (Memberships, t-shirts, etc.) and Benefitable allows for different kinds of beneficiaries (Reservation-holders, t-shirt recipients, honorees for donations, etc) that don't have to be ThemeContacts.  

It's just that I didn't know how to engineer for those things more specifically when those things are still notional.  So I just tried to build space into the objects and the processes by which they're handled for those other things, once they become real.